components/omega/src/CMakeLists.txt

```diff
@@ -3,44 +3,46 @@
 # Add source files for the library
 file(GLOB _LIBSRC_FILES infra/*.cpp base/*.cpp ocn/*.cpp)
 
-# Create the library target
 add_library(${OMEGA_LIB_NAME} ${_LIBSRC_FILES})
 
-# add include directories
 target_include_directories(
     ${OMEGA_LIB_NAME}
-    PRIVATE
-    ${OMEGA_SOURCE_DIR}/src/base
-    ${OMEGA_SOURCE_DIR}/src/infra
-    ${Parmetis_INCLUDE_DIRS}
-)
-
-# add compiler options
-target_include_directories(
-    ${OMEGA_LIB_NAME}
-    PRIVATE
+    PUBLIC
     ${OMEGA_SOURCE_DIR}/src/base
     ${OMEGA_SOURCE_DIR}/src/infra
     ${OMEGA_SOURCE_DIR}/src/ocn
+    ${Parmetis_INCLUDE_DIRS}
 )
 
-target_compile_options(
+target_compile_definitions(
     ${OMEGA_LIB_NAME}
-    PRIVATE
-     ${OMEGA_CXX_FLAGS}
+    PUBLIC
+    OMEGA_ARCH=${OMEGA_ARCH}
 )
 
-# add linker options
 target_link_options(
     ${OMEGA_LIB_NAME}
-    PRIVATE
+    PUBLIC
     ${OMEGA_LINK_OPTIONS}
 )
 
-target_link_libraries(${OMEGA_LIB_NAME} spdlog yakl pioc yaml-cpp parmetis metis)
+target_link_libraries(
```

Comment:
Any reason (here as well as other parts of this file) not to use PRIVATE instead of PUBLIC? I didn't think downstream customers are interested in getting all the tpls compiler/linker flags..

Reply:
You can use interface targets to take care of Omega specific flags. You can do something like

```cmake
add_library(Omega_flags INTERFACE) 
target_compile_options(Omega_flags PUBLIC ${MY_FLAGS})
target_compile_definitions(Omega_flags PUBLIC ${MY_CPP_DEFS})
#also link flags if you need to
```
Then you link this library _privately_ to the Omega lib, as well as to all the unit tests. `e3sm.exe` will never see the flags, since this dummy target is linked privately to Omega. Also, at build time, nothing gets built for this library (it is a bit of an abuse of language to call it a library).

This is a standard technique in cmake to pass flags/macros to multiple downstream targets, without using variables (other than for creating the interface target)

Reply:
I updated Omega script based on the suggestion. Thank you!